cpp/src/parquet/metadata.cc

```diff
@@ -793,26 +793,29 @@ class FileMetaData::FileMetaDataImpl {
 
     std::shared_ptr<FileMetaData> out(new FileMetaData());
     out->impl_ = std::make_unique<FileMetaDataImpl>();
-    out->impl_->metadata_ = std::make_unique<format::FileMetaData>();
-
-    auto metadata = out->impl_->metadata_.get();
-    metadata->version = metadata_->version;
-    metadata->schema = metadata_->schema;
+    {
+      // GH-39336: Copy the metadata, but only the row groups we need.
```

Comment:
Here it might looks a bit hacking, this avoid copy the rowGroup in `format::FileMetaData::FileMetaData`.